examples/vhdl/array_axis_vcs/runall_addwave.do

```diff
@@ -1,5 +1,5 @@
 echo "Running DO File"
-foreach v "vunit_axism vunit_axiss uut uut/fifo" {
+foreach v "uut_vc/vunit_axism uut_vc/vunit_axiss uut_vc/uut uut_vc/uut/fifo" {
   add wave -group "$v" -position insertpoint sim:/tb_axis_loop/$v/*
 }
 run -all;
```

Comment:
I would prefer if you remove this line such that the user has to run manually when in GUI mode. When this test is run without the `--gui` option, VUnit will pass the follow script lines to the simulation:

```
set failed [vunit_load]
if {$failed} {quit -code 1}
set failed [vunit_run]
if {$failed} {quit -code 1}
quit -code 0
```

The  `vunit_load` command will load the design and then source `runall_addwave.do`. That script runs the test once with `run -all` until it completes. Then, `vunit_run` will continue to run after the stopping point which will continue forever, just toggling the clock, and the call to `run.py` will never return,

Reply:
Hi @LarsAsplund!

As you mentioned, the content of the script `batch.do`, located at `vunit_out/test_output/lib_name.tb_name.hash/modelsim`, is generated with the following content:

```bash
onerror {quit -code 1}
source "/home/.../vunit/examples/vhdl/array_axis_vcs/vunit_out/test_output/lib.tb_axis_loop.test_260bb5c8c675e898eca5dc9024a4420ede12c0bc/modelsim/common.do"
set failed [vunit_load]
if {$failed} {quit -code 1}
set failed [vunit_run]
if {$failed} {quit -code 1}
quit -code 0
```

However, this is not only the case when you run it without the `-g` option. I’ve performed a couple of tests, and this script remains the same when running with the `-g ` option, with `VU.set_sim_option('modelsim.init_files.after_load', ['runall_addwave.do'])` commented out, or even in other examples I tested. This makes me think that the `batch.do` file is "always" generated with this content. :thinking:

Nevertheless, you are absolutely right, this simulation (even before my change) freezes when you run it without the `-g` option (after the test's verbose output), as shown in the last image of the main comment. In fact, I had to skip the simulation to stop it.

I solved this "freeze" issue by adding `quit -code 1;` at the end of the `runall_addwave.do script. The updated script is as follows:

``` bash
echo "Running DO File"
foreach v "uut_vc/vunit_axism uut_vc/vunit_axiss uut_vc/uut uut_vc/uut/fifo" {
  add wave -group "$v" -position insertpoint sim:/tb_axis_loop/$v/*
}
run -all;
quit -code 1;
``` 

I've verified that with the `-g` option, the Questa GUI opens with the waveform, and without this option, the simulation finishes correctly.

If you have a couple of minutes, could you verify the correct operation of this change? Once confirmed, I will add it to the PR.

Thanks for the feedback, Lars!

Cheers!

Reply:
Correct execution of the test (after adding `quit -code 1;`) without the `-g` option:

![vunit_PR_5](https://github.com/user-attachments/assets/21ea4edd-4e1a-416d-bf8d-cfadba1baf4f)

Reply:
Yes, `batch.do` is the same in both cases but `vunit_load` defined in `common.do` differs. In one case it calls `runall_addwave.do` but not in the other.

Wouldn't a run with the `-g` option exit immediately after the simulation ended if you add `quit` to the script?

Reply:
> Yes, batch.do is the same in both cases but vunit_load defined in common.do differs. In one case it calls runall_addwave.do but not in the other.

Oh, I see!

> Wouldn't a run with the -g option exit immediately after the simulation ended if you add quit to the script?

Oh, my bad! You are absolutely right. I initially checked that Questa opened, but I closed it immediately. Now, I’ve checked again, and if you leave it open for some time, the GUI does eventually close. :disappointed:

Sorry about that! I’ll look for another solution.

Reply:
Since the DO file is only relevant to the GUI mode it would be better to use this flag instead:

```
VU.set_sim_option("modelsim.init_file.gui", "runall_addwave.do")
```

No need to comment out that line.

Reply:
🫡